corehq/ex-submodules/pillowtop/checkpoints/manager.py

```diff
@@ -123,16 +123,19 @@ def __init__(self, checkpoint, checkpoint_frequency, checkpoint_callback=None):
         self.max_checkpoint_delay = getattr(settings, 'PTOP_CHECKPOINT_DELAY_OVERRIDE', MAX_CHECKPOINT_DELAY)
         self.checkpoint = checkpoint
         self.checkpoint_frequency = checkpoint_frequency
-        self.last_update = datetime.utcnow()
-        self.last_log = datetime.utcnow()
+        self.last_update = None
+        self.last_log = None
         self.checkpoint_callback = checkpoint_callback
 
     def should_update_checkpoint(self, context):
         frequency_hit = context.changes_seen >= self.checkpoint_frequency
         time_hit = False
         if self.max_checkpoint_delay:
-            seconds_since_last_update = (datetime.utcnow() - self.last_update).total_seconds()
-            time_hit = seconds_since_last_update >= self.max_checkpoint_delay
+            if self.last_update is not None:
+                seconds_since_last_update = (datetime.utcnow() - self.last_update).total_seconds()
+                time_hit = seconds_since_last_update >= self.max_checkpoint_delay
+            else:
+                time_hit = True
```

Comment:
`self.last_update` is `None` when `PillowCheckpointEventHandler` has just been initialized. Does that mean we always want to write a checkpoint immediately when its initialized, which I assume is when the process starts up? Or does `should_update_checkpoint` only get called after some period of time, and it knows that (and relies on it)?

Reply:
Yeah, I thought checkpointing on startup was a nice way to ensure periodic checkpoints.